2015/presentations/locustio_load_testing_tool/text.md

```diff
@@ -0,0 +1,252 @@
+# Locust.io - nowoczesne narzędzie do load testów - Jacek Nosal
+
+
+## Wprowadzenie
+
+
+Testowanie obciążenia aplikacji webowych to temat bardzo złożony, który nabiera
+na znaczeniu, jeżeli pod uwagę weźmiemy skalowalne architektury. Dzisiejszy
+użytkownik wymaga. Coraz częściej spotyka się statystyki, które pokazują jak
+wiele użytkowników rezygnuje ze skorzystania z danej aplikacji, jeżeli czasy 
+odpowiedzi są wyższe niż pewien (niski) próg. Im szybsze odpowiedzi, tym więcej
+użytkowników wejdzie na naszą stronę (ponieważ szybciej będą w stanie ocenić 
+czy znajduje się na niej to, czego szukają). Typowy odwiedzający nie dba również
+o stronę techniczną, bo po prostu się an tym nie zna. Aplikacja może cachować
+dużo danych, autoscalować się, zepewniać failover - to wszystko nie ma wielkiego
+znaczenia jeżeli jest po prostu wolne. Naturalnym wydaje się więc posiadanie
+możliwości symulacji dużego ruchu, który sprawdzi możliwości naszej aplikacji 
+i pokaże jak cały stack sprawuje się gdy zostanie "nawiedzony" przez tysiące
+użytkowników jednocześnie. 
+
+
+Spośród wielu dostępnych narzędzi na rynku, jedno wyróżnia się szczególnie -
+locust.io. Po pierwsze: napisane jest w Pythona. Po drugie: do jego obsługi
+wykorzystujemy Pythona. Po trzecie: jak większość frameworków jest niesamowicie
+elastyczny i stworzony tak, żeby zmiana coreowych funkcjonalności nie wydawała
+się czymś magicznym.
+
+
+## Locust.io
+
+
+Locust to lekkie narzędzie, które umożliwia stworzenie w Pythonie scenariuszy
+do interakcji z aplikacją, które będą użyte do symulacji użytkowników końcowych.
+Narzędzie potrafi symulować praktycznie dowolną liczbę znajdujących się
+w osobnych sesjach użytkowników.
+
+Od strony technicznej narzędzie to spójne połączenie flaska, geventa oraz pyzmq.
+Całość została tak zaprojektowana, żeby umożliwić rozdystrybuowanie testów na 
+kilka maszyn.
+
+
+Do zalet locusta należą:
+
+* możliwość pisania scenariuszy testowych w Pythonie
+* skalowalność
+* panel webowy do kontrolowania procesu testowania
+* możliwość do przetestowania każdego systemu (nie tylko tych, opierających
+ się na HTTP)
+
+
+Narzędzie to rozwiązuje jeszcze jeden problem: proces testowania jest po prostu
```

Comment:
Moje pierwsze odczucie po przeczytaniu tego zdania było takie że to w locust.io proces testowania jest nudny i żmudny. Proponuję przeredagować.